crates/pera-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -1069,38 +1082,42 @@ impl AuthorityPerEpochStore {
         let encryption_of_decryption_key_shares = encryption_of_decryption_key_shares
             .contents
             .into_iter()
-            .map(|entry| (entry.key, entry.value))
-            .collect();
+            .map(|entry| Ok((DWalletMPCNetworkKey::try_from(entry.key)?, entry.value)))
+            .collect::<DwalletMPCResult<HashMap<_, _>>>()?;
+
         Ok(encryption_of_decryption_key_shares)
     }
 
-    pub fn committee_validators_class_groups_public_keys_and_proofs(
+    /// Reads the validator's decryption key share for the current epoch from all key versions , if exists in the system state.
+    ///
+    /// The data is loaded from the epoch start system state. The returned value is a map where:
+    /// - The key represents the key scheme as a `u8` identifier.
+    /// - The value is a vector of vectors (`Vec<Vec<u8>>`), containing the decryption key shares.
+    pub fn load_decryption_key_shares_from_system_state(
         &self,
-    ) -> DwalletMPCResult<HashMap<PartyID, ClassGroupsPublicKeyAndProof>> {
-        let public_keys_and_proofs = match self.epoch_start_state() {
-            EpochStartSystemState::V1(data) => {
-                let committee: Vec<_> = self
-                    .committee()
-                    .voting_rights
-                    .iter()
-                    .map(|(name, _)| name.clone())
-                    .collect();
+    ) -> DwalletMPCResult<HashMap<DWalletMPCNetworkKey, Vec<Vec<u8>>>> {
+        let encryption_of_decryption_key_shares =
+            self.load_encryption_of_decryption_key_shares_from_system_state()?;
+        let party_id = authority_name_to_party_id(&self.name, self)? as usize;
```

Comment:
make it clear from the docs that this function return only "mine" (from the validator prespective) shares. also change to pub crate